File: Esay/Move Zeroes.java
//https://leetcode.com/problems/move-zeroes/description/
import java.util.*;
class Solution {
    public void moveZeroes(int[] nums) {
        int a=-1;
        int n =nums.length;
        for(int i=0;i<n;i++){
            if(nums[i]!=0){
                swap(nums,++a,i);
            }
        }
    }
    private void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
}

File: Esay/Move Zeroes.cpp
//https://leetcode.com/problems/move-zeroes/description/
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
    int lastNonZeroFoundAt = -1;
    int n = nums.size();

    for (int currentIndex = 0; currentIndex < n; ++currentIndex) {
        if (nums[currentIndex] != 0) {
            swap(nums[++lastNonZeroFoundAt], nums[currentIndex]);
        }
    }
  }
};
